internal/app/collector/monitors/missing_votes_governance_monitor.go

```diff
@@ -0,0 +1,182 @@
+	return nil
+}
+
+func (m *MissingVotesGovernanceMonitor) GetMetricVectors() map[MetricName]*MetricVector {
+	m.lock.RLock()
+	defer m.lock.RUnlock()
+	return m.metricVectors
+}
+
+func (m *MissingVotesGovernanceMonitor) Handler(ctx context.Context) error {
+	tmpMetricVectors := make(map[MetricName]*MetricVector)
+	initMetrics(nil, m.providedMetricVectors(), nil, tmpMetricVectors)
+
+	notVotedValidators := make(map[string]uint)
+
+	proposalList, err := m.FetchLastProposals(ctx)
+	if err != nil {
+		m.logger.Errorf("Could not get proposalList. Error: %v", err)
+		return err
+	}
+
+	// Aggregate not voted validators counts
+	for _, item := range proposalList {
+		proposalID, err := strconv.Atoi(*item.ID)
+		if err != nil {
+			m.logger.Errorf("Could not convert ID to integer. ID: %v", item.ID)
+			return err
+		}
+
+		notVotedForProposal, err := m.FetchNotVotedValidators(ctx, proposalID)
+		if err != nil {
+			m.logger.Errorf("Could not get proposal for %v. Error: %v", item.ID, err)
+			return err
+		}
+
+		for _, validatorItem := range notVotedForProposal {
+			notVotedValidators[validatorItem] += 1
+		}
+	}
+
+	for address, notVotedCount := range notVotedValidators {
+		tmpMetricVectors[MissedVotesMetric].Set(address, float64(notVotedCount))
+		tmpMetricVectors[AlertLimitMetric].Set(address, float64(m.alertLimit))
+	}
+
+	m.logger.Infof("Successfully retrieved list of missing votes")
+
+	m.lock.Lock()
+	defer m.lock.Unlock()
+	copyVectors(tmpMetricVectors, m.metricVectors)
+
+	return nil
+}
+
+func (m *MissingVotesGovernanceMonitor) InitMetrics() {
+	initMetrics(nil, m.providedMetricVectors(), nil, m.metricVectors)
+}
+
+func (m *MissingVotesGovernanceMonitor) providedMetricVectors() []MetricName {
+	return []MetricName{MissedVotesMetric, AlertLimitMetric}
+}
+
+func (m *MissingVotesGovernanceMonitor) FetchLastProposals(ctx context.Context) ([]*models.GetProposalListResultProposals, error) {
+	resp, err := m.apiClient.Governance.GetV1GovProposals(
+		&governance.GetV1GovProposalsParams{
+			Context: ctx,
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get proposals with {%s}", err)
+	}
+
+	err = resp.GetPayload().Validate(nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to validate proposals with {%s}", err)
+	}
+
+	proposalsList := resp.GetPayload().Proposals
+
+	result := make([]*models.GetProposalListResultProposals, 0)
+
+	for _, item := range proposalsList {
+		if *item.Status == "Passed" || *item.Status == "Rejected" {
+			result = append(result, item)
+		}
+
+		if len(result) >= m.lookbackLimit {
+			break
+		}
+	}
+
+	return result, nil
+}
+
+func (m *MissingVotesGovernanceMonitor) FetchNotVotedValidators(ctx context.Context, proposalID int) ([]string, error) {
+	res := make([]string, 0)
+
+	proposalVotes, err := m.repository.GetVotes(ctx, proposalID)
+	if err != nil {
+		return res, fmt.Errorf("failed to get votes with {%s}", err)
+	}
+
+	votedValidatorsSubset := make([]string, 0)
```

Comment:
`votedValidatorsSubset := make([]string, 0, len(proposalVotes))`
when the result slice size is known, it's a good practice the allocate the slice of that size